javascript/javascript3/week2/visual-promise/main.js

```diff
@@ -0,0 +1,19 @@
+//Using promise
+const liArray = document.querySelectorAll('.marks >li');
+const translateOneByOne =() => {
+    moveElement(liArray[0], { x: 20, y: 300 }).then(() => {
+        console.log("Red element has been moved");
+    })
+        .then(() => moveElement(liArray[1], { x: 400, y: 300 }))
+        .then(() => console.log("Blue element has been moved"))
+        .then(() => moveElement(liArray[2], { x: 400, y: 20 }))
+        .then(() => console.log("Green element has been moved"))    
+}
+translateOneByOne();
+
+//Using promiseAll
+function translateAllAtOnce() {
+    Promise.all([moveElement(liArray[0], { x: 20, y: 300 }), moveElement(liArray[1], { x: 400, y: 300 }), moveElement(liArray[2], { x: 400, y: 20 })]);
+    console.log('all done');
```

Comment:
Here, the `console.log()` method is executed before `Promise.all()` is finished and all the elements are moved. You can use `console.log()` inside `then()` to make it wait for all the functions inside `Promise.all()` are executed -or you can use the other method that you learned from the class :)-